src/data/repository.py

```
@@ -0,0 +1,267 @@
+    def delete_asset(self, asset_id: str) -> None:
+        """Delete an asset and cascading relationships/events."""
+
+        asset = self.session.get(AssetORM, asset_id)
+        if asset is not None:
+            self.session.delete(asset)
+
+    # ------------------------------------------------------------------
+    # Relationship helpers
+    # ------------------------------------------------------------------
+    def add_or_update_relationship(
+        self, source_id: str, target_id: str, rel_type: str, strength: float, bidirectional: bool
```

Ruff (0.14.3)

Boolean-typed positional argument in function definition

(FBT001)